Formula/janet.rb

```diff
@@ -18,15 +18,30 @@ class Janet < Formula
   depends_on "meson" => :build
   depends_on "ninja" => :build
 
+  resource "jpm" do
+    url "https://github.com/janet-lang/jpm/archive/refs/tags/v1.0.0.tar.gz"
+    sha256 "858d4ef2f6ac78222c53154dd91f8fb5994e3c3cbe253c9b0d3b9d52557eeb9b"
+  end
```

Comment:
Why not a separate formula?

Reply:
It used to be bundled with `janet` so it's a breaking change for existing users to have it as two separate formula

Reply:
Upstream seems to have decided to do that, and presumably had good reasons for doing so despite it being a breaking change, so I don't think that reason is sufficient for us not have this as a separate formula.

Reply:
Upstream isn't a breaking change for manual installation, all that's different is where [git clones from](https://janet-lang.org/docs/jpm.html) in the first step.

It's a similar situation to the `nodejs` formula bootstrapping `npm`. You could have them separated but the majority of use cases don't make sense without `jpm`.

I don't have any experience making new formula, but I can give it a go if you think this is the wrong approach.

Reply:
> It used to be bundled with janet so it's a breaking change for existing users to have it as two separate formula

I'm not sure why that would be a breaking change. If people install it they'll still get jpm because it will be a dependency.

> It's a similar situation to the nodejs formula bootstrapping npm. You could have them separated but the majority of use cases for janet don't make sense without jpm.

But upstream clearly thinks there are jpm usecases without janet, so that's why they should be separate formula

Reply:
`npm` and `node` are bundled because their versions are tightly coupled -- we almost never update `npm` without also updating `node`. It also helps a lot that the Node devs also bundle `npm` with the `node` source tarball.

If `jpm`'s version is as tightly coupled with `janet`'s, then it might make sense to do the same here. But if it's likely that we want to update `jpm` without updating `janet`, then `jpm` really ought to be a separate formula.